cmd/byohctl/service/agent.go

```diff
@@ -0,0 +1,602 @@
+
+				// Log the error
+				utils.LogWarn("Failed to install %s: %v", pkg.name, err)
+			} else {
+				// Log success
+				utils.LogSuccess("%s installed successfully", pkg.name)
+				installedPackages = append(installedPackages, pkg.name)
+			}
+
+			// Verify installation
+			if pkg.packageName != "" && !checkPackageInstalled(pkg.packageName) {
+				utils.LogWarn("%s installation verification failed", pkg.name)
+				installationErrors = append(installationErrors,
+					fmt.Errorf("%s installation verification failed", pkg.name))
+			}
+		} else {
+			// For packages that are already installed, log at debug level to reduce noise
+			utils.LogDebug("%s is already installed", pkg.name)
+		}
+	}
+
+	// If we encountered errors, return a combined error
+	if len(installationErrors) > 0 {
+		errorMessages := []string{}
+		for _, err := range installationErrors {
+			errorMessages = append(errorMessages, err.Error())
+		}
+		return fmt.Errorf("package installation errors: %s", strings.Join(errorMessages, "; "))
+	}
+
+	// Only log a message if we actually installed something
+	if len(installedPackages) > 0 {
+		utils.LogSuccess("All required packages installed successfully")
+	}
+
+	return nil
+}
+
+var requiredPackages = []struct {
+	name         string               // Package name
+	installCmd   string               // Installation command
+	installMsg   string               // Installation message
+	packageName  string               // Debian package name (if different from name)
+	versionCheck func() (bool, error) // Optional function to verify version
+}{
+	{
+		name:       "imgpkg",
+		installCmd: "curl -s -L https://carvel.dev/install.sh | bash",
+		installMsg: "Installing imgpkg using Carvel install script",
+	},
+	{
+		name:        "dpkg",
+		installCmd:  "apt-get update && apt-get install -y dpkg",
+		installMsg:  "Installing dpkg package manager",
+		packageName: "dpkg",
+	},
+	{
+		name:        "ebtables",
+		installCmd:  "apt-get update && apt-get install -y ebtables || apt-get install -y --fix-broken && apt-get install -y ebtables || apt-get install -y --reinstall ebtables",
+		installMsg:  "Installing ebtables Ethernet bridge filtering",
+		packageName: "ebtables",
+	},
+	{
+		name:        "conntrack",
+		installCmd:  "apt-get update && apt-get install -y conntrack || apt-get install -y --fix-broken && apt-get install -y conntrack",
+		installMsg:  "Installing conntrack network connection tracking",
+		packageName: "conntrack",
+	},
+	{
+		name:        "socat",
+		installCmd:  "apt-get update && apt-get install -y socat || apt-get install -y --fix-broken && apt-get install -y socat",
+		installMsg:  "Installing socat multipurpose relay",
+		packageName: "socat",
+	},
+	{
+		name:         "libseccomp2",
+		installCmd:   "apt-get install -y --no-install-recommends libseccomp2",
+		installMsg:   "Installing libseccomp2 library",
+		packageName:  "libseccomp2",
+		versionCheck: checkLibseccompVersion,
+	},
+}
```

Comment:
Jayanth, the installation process for these packages are inconsistent. Why do some of them have multiple installation methods, but others simply have a `apt-get install`?

Reply:
Sure... chnaged to a consistent approach in the recent commit